src/compiler/transformers/declarations/emitBinder.ts

```diff
@@ -0,0 +1,914 @@
+            const symbol = createEmitSymbol();
+            symbol.declarations.push(node);
+            setValueDeclaration(symbol as Symbol, node);
+            node.symbol = symbol as Symbol;
+            return symbol;
+        }
+        function createEmitSymbol(): EmitDeclarationSymbol {
+            return {
+                parent: currentSymbol,
+                declarations: [],
+                flags: 0,
+            };
+        }
+        function getSymbol(table: EmitDeclarationSymbolTable, name: MemberKey) {
+            let symbol = table.get(name);
+            if (!symbol) {
+                symbol = createEmitSymbol();
+                symbol.name = name;
+                table.set(name, symbol);
+            }
+            return symbol;
+        }
+        function addLocalAndExportDeclaration(name: LocalAndExportName | MemberKey | undefined, node: Declaration, flags: SymbolRegistrationFlags, isExport: boolean, isBlockScoped = true) {
+            const { exportName, localName } = typeof name === "object" ? name : { exportName: name, localName: name };
+            if (isExport) {
+                const exportKind = flags.includes & SymbolFlags.Value ? SymbolFlags.ExportValue : 0;
+                const localSymbol = addLocalOnlyDeclaration(localName, node, { includes: exportKind, excludes: flags.excludes }, isBlockScoped);
+                const exportSymbol = addExportOnlyDeclaration(exportName, node, flags);
+                localSymbol.exportSymbol = exportSymbol;
+                // Export symbol can be undefined if the export modifier was placed in an unexpected position.
+                // We just assume the local symbol should be used. There are already bigger issues in the file anyway.
+                return exportSymbol ?? localSymbol;
+            }
+            else {
+                return addLocalOnlyDeclaration(localName, node, flags, isBlockScoped);
+            }
+        }
+        function addExportOnlyDeclaration(name: MemberKey | undefined, node: Declaration, flagsAndForbiddenFlags: SymbolRegistrationFlags) {
+            if (!currentExportsSymbolTable) {
+                return undefined;
+            }
+            return addDeclaration(currentExportsSymbolTable, name, node, flagsAndForbiddenFlags);
+        }
+        function addLocalOnlyDeclaration(name: MemberKey | undefined, node: Declaration, flagsAndForbiddenFlags: SymbolRegistrationFlags, isBlockScoped = true) {
+            return addDeclaration(isBlockScoped ? currentLocalSymbolTable : currentFunctionLocalSymbolTable, name, node, flagsAndForbiddenFlags);
+        }
+
+        function addDeclaration(table: EmitDeclarationSymbolTable, name: MemberKey | undefined, node: Declaration, { includes, excludes }: SymbolRegistrationFlags) {
+            let symbol = name !== undefined ? getSymbol(table, name) : createEmitSymbol();
+            // Symbols don't merge, create new one

```

Comment:
The binder issues an errors in this case and merges anyway. How does creating a new symbol affect emit?

Reply:
I don't think it does always merge. From what I understand [here](https://github.com/microsoft/TypeScript/blob/main/src/compiler/binder.ts#L872), the binder will create a new symbol if `symbol.flags & excludes`. I did spend quite a bit of time trying to emulate the same emit behavior, but I'm ultimately not sure how much this matters anyway since these will be error cases and emulating teh same behavior in an error is not crucial